www/include/configuration/configObject/timeperiod/DB-Func.php

```diff
@@ -475,3 +465,104 @@ function testTemplateLoop($value)
 
     return true;
 }
+
+/**
+ * All in one function to duplicate time periods
+ *
+ * @param array $params
+ * @return int
+ */
+function duplicateTimePeriod(array $params): int
+{
+    global $pearDB;
+
+    $isAlreadyInTransaction = $pearDB->inTransaction();
+    if (!$isAlreadyInTransaction) {
+        $pearDB->beginTransaction();
+    }
+    try {
+        $params['tp_id'] = createTimePeriod($params);
+        createTimePeriodsExceptions($params);
+        createTimePeriodsIncludeRelations($params);
+        createTimePeriodsExcludeRelations($params);
+        if (!$isAlreadyInTransaction) {
+            $pearDB->commit();
+        }
+    } catch (\Exception $e) {
+        if (!$isAlreadyInTransaction) {
+            $pearDB->rollBack();
+        }
+    }
+    return $params['tp_id'];
+}
+
+/**
+ * Creates time period and returns id.
+ *
+ * @param array $params
+ * @return int
+ */
+function createTimePeriod(array $params): int
+{
+    global $pearDB;
+
+    $pearDB->query("INSERT INTO timeperiod VALUES (" . $params['values'] . ")");
+    return $pearDB->lastInsertId();
```

Comment:
```suggestion
    return (int) $pearDB->lastInsertId();
```

If you cast here, you don't have to cast (int) $params['tp_id'] in your other following methods,
then your @return int in your phpdoc will be correct :), as lastInsertId() return a string